test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -396,6 +396,53 @@
             #"shutdown-on-error triggered because of exception"
             :error))))))
 
+(deftest ^:integration debouncing-test
+  (testing "Default debouncing behavior"
+    (let [root-dir (fs/temp-dir "debounce")
+          actual-events (atom [])
+          actual-calls (atom 0)
+          callback (fn [e]
+                     (swap! actual-events concat e)
+                     (swap! actual-calls inc))
+          app (tk/boot-services-with-config watch-service-and-deps {})
+          svc (tk-app/get-service app :FilesystemWatchService)
+          event-cadence (quot watch-core/window-min 2)]
+      ;; We trigger half the number of events that could independently occur
+      ;; without exceeding the window-max, we trigger them at half the wait
+      ;; of window-min to ensure they will all be caught even if the host
+      ;; running the tests is slow. This requires that window-max be at least
+      ;; twice the duration as window-min.
```

Comment:
It increases the likelihood.

I definitely think this test has race issues it. I'm not sure the best way to test this - a quiet period and a timeout in a background thread at the integration level - without there being a race condition. My hope was to make the temporal relationships and assumptions explicit and fail the test in a more obvious way if the test runners performance goes out of the bands where this test would perform reliably.

My assumption when writing this test was that if we were seeing slowdowns large enough that something triggered at half our window-min could exceed our window-max we would either be seeing slow downs that would affect other tests (ie if what should happen in 50ms was taking over 2s we were in some kind of failed state on that host and we probably shouldn't trust our results elsewhere either) or our default settings are poorly chosen (ie we can't really test this reliably if we set our window-min to 2ms and our window-max to 8ms when we expect events to reach us within 10ms of when they occur).